client/src/Dataset/Details/details.hooks.tsx

```diff
@@ -0,0 +1,174 @@
+
+    // actions
+    const generateAction = async () => {
+        // TODO
+    };
+    const scanAction = async () => {
+        // TODO
+    };
+    const importAction = async () => {
+        // TODO
+    };
+    const exportAction = async () => {
+        // TODO
+    };
+    const deleteImageAction = async (id: number) => {
+        await DatasetApi.deleteImage(id);
+        details.refreshPage();
+    };
+    const downloadImageAction = (id: number) => {
+        // TODO
+        details.refreshPage();
+    };
+    const resetMetadataAction = () => {
+        // TODO
+    };
+
+    return {
+        tabs,
+        filters,
+        folder,
+        details,
+        actions: {
+            generateAction,
+            scanAction,
+            importAction,
+            exportAction,
+            deleteImageAction,
+            annotateImageAction: openAnnotator,
+            downloadImageAction,
+            resetMetadataAction,
+        },
+    };
+};
+
+const useFilterState = (): FilterState => {
+    const contains = useState<string>('');
+    const order = useState<string>('File Name');
+    const annotatedOn = useState<boolean>(false);
+    const notAnnotatedOn = useState<boolean>(false);
+
+    return {
+        contains,
+        order,
+        annotatedOn,
+        notAnnotatedOn,
+    };
+};
+
+const useDetails = (id: number): DetailsState => {
+    const [, dispatch] = useGlobalContext();
+    const [page, setPage] = useState(0);
+    const [generation, moveGeneration] = useState(0);
+
+    const [dataset, setDataset] = useState<Dataset | null>(null);
+    const [images, setImages] = useState<Image[]>([]);
+    const [categories, setCategories] = useState<string[]>([]);
+    const [imagesCount, setImagesCount] = useState<number>(0);
+    const [pagesCount, setPagesCount] = useState<number>(0);
+    const [subdirectories, setSubdirectories] = useState<string[]>([]);
+
+    const refreshPage = () => {
+        moveGeneration(c => c + 1);
+    };
+
+    useEffect(() => {
+        const update = async () => {
+            const process = 'Loading images from dataset';
+
+            addProcess(dispatch, process);
+
+            // TODO add folder order param
+            const { data } = await DatasetApi.getDetails(id);
+            setDataset(data.dataset);
+            setImages(data.images);
+            setCategories(data.categories);
+            setImagesCount(data.total);
+            setPagesCount(data.pages);
+            setSubdirectories(data.subdirectories);
+
+            removeProcess(dispatch, process);
+        };
+        update();
+    }, [id, page, generation, dispatch]);
+
+    return {
+        dataset,
+        images,
+        categories,
+        imagesCount,
+        pagesCount,
+        subdirectories,
+        page,
+        setPage,
+        refreshPage,
+    };
+};
+
+const usePageNavigation = (): NavigationState => {
+    const navigation = useNavigation();
+    const openAnnotator = (id: number) => {
```

Comment:
this function is static against its arguments you don't have to recreate it every time you call `usePageNavigation`. Move it to outer scope.